src/core/i18n/utils.js

```diff
@@ -80,3 +80,17 @@ export function isRtlLang(lang) {
 export function getDirection(lang) {
   return isRtlLang(lang) ? 'rtl' : 'ltr';
 }
+
+export function getLangFromRouter(renderProps) {
+  let lang = defaultLang;
+  if (renderProps && renderProps.params) {
+    // Get the lang from the url param by default
+    // if it exists.
+    if (renderProps.params.lang) {
+      lang = getLanguage(renderProps.params.lang);
+    } else if (renderProps.params.query && renderProps.params.query.lang) {
```

Comment:
I think this should be `renderProps.location.query.lang` instead. I think I used this in my comment before but I was incorrect.